### PR TITLE
Print application logs when deployment fails

### PR DESCRIPTION
**Pull request by RuoyuWang-MS**
When spring app deploy failure, we should print the app logs for customer diagnosing.

### CODE REVIEW

File: src/DeploymentProvider/DeploymentHelper.ts [hunk 1]
```diff
@@ -422,4 +418,55 @@ export class DeploymentHelper {
...
+        logParams['limitBytes'] = 1024 * 1024;
+        logParams['sinceSeconds'] = 300;
+        logParams['follow'] = true;
+        const credentials = Buffer.from(`primary:${logStream["primaryKey"]}`).toString('base64');
+        const auth = { "Authorization" : `Basic ${credentials}` };
+        var url = new URL(streamingUrl)
+        url.search = new URLSearchParams(logParams).toString();
+        fetch(url, {method: 'GET', headers : auth })
+            .then(response => {
+            if (response.ok) {
```
Comment (allxiao) [inline src/DeploymentProvider/DeploymentHelper.ts:448]: nit: it seems the indentation here is broken.

File: src/DeploymentProvider/DeploymentHelper.ts [hunk 2]
```diff
@@ -422,4 +418,55 @@ export class DeploymentHelper {
...
+            console.error('Error:', error.message);
+          });;
+    }
+
+    private static async deployWithLog(client: asa.AppPlatformManagementClient, params: ActionParameters, deploymentResource: asa.DeploymentResource) {
+        try {
+            const response = await client.deployments.beginCreateOrUpdateAndWait(params.resourceGroupName, params.serviceName, params.appName, params.deploymentName, deploymentResource);
+            core.debug('deploy response: ' + JSON.stringify(response));
+        } catch (e:any) {
+            await this.printLatestAppInstanceLog(client, params);
```
Comment (allxiao) [inline src/DeploymentProvider/DeploymentHelper.ts:468]: I suggest we first output some hint message that we met some exception (maybe log the exception first?), and then print some message that we will try to stream the latest app logs to the console. Then start dumping the logs.

What's the behavior in CLI?

Comment (RuoyuWang-MS) [reply src/DeploymentProvider/DeploymentHelper.ts:469]: CLI will print app logs and throw the error at the end. 
image

I add one hint message for users to check the app log.
image

File: src/DeploymentProvider/DeploymentHelper.ts (hunk 1 above)
Comment (RuoyuWang-MS) [reply src/DeploymentProvider/DeploymentHelper.ts]: Fixed

**Review by allxiao — APPROVED** [review]

**Pull request merged**